R/mod_indicator.R

```diff
@@ -0,0 +1,400 @@
+      )
+    })
+
+    output$set_include <- shiny::renderUI({
+      shiny::req(input$indicator)
+      shiny::tags$div(
+        title = "Angi om indikatoren skal vises frem eller ikke",
+        shiny::checkboxInput(
+          ns("include"), "Vis i Sykehusviseren",
+          value = as.logical(rv$ind_data$include)
+        )
+      )
+    })
+
+    output$set_level_direction <- shiny::renderUI({
+      shiny::req(input$indicator)
+      shiny::tags$div(
+        title = paste(
+          "Ta vekk haken hvis synkende indikatorverdier gir \u00f8kt",
+          "m\u00e5loppn\u00e5else"
+        ),
+        shiny::checkboxInput(
+          ns("level_direction"),
+          "H\u00f8y verdi gir h\u00f8y m\u00e5loppn\u00e5else",
+          value = as.logical(rv$ind_data$level_direction)
+        )
+      )
+    })
+
+    output$set_level_green <- shiny::renderUI({
+      shiny::req(input$indicator)
+      shiny::tags$div(
+        title = "Grenseverdi for beste m\u00e5loppn\u00e5else",
+        shiny::numericInput(
+          ns("level_green"),
+          paste("H\u00f8y m\u00e5loppn\u00e5else", rv$level_logi),
+          value = rv$ind_data$level_green,
+          min = rv$level_green_min,
+          max = rv$level_green_max,
+          step = 0.1
+        )
+      )
+    })
+
+    output$set_level_yellow <- shiny::renderUI({
+      shiny::req(input$indicator)
+      shiny::tags$div(
+        title = "Grenseverdi for middels m\u00e5loppn\u00e5else",
+        shiny::numericInput(
+          ns("level_yellow"),
+          paste("Middels m\u00e5loppn\u00e5else", rv$level_logi),
+          value = rv$ind_data$level_yellow,
+          min = rv$level_yellow_min,
+          max = rv$level_yellow_max,
+          step = 0.1
+        )
+      )
+    })
+
+    output$set_min_denominator <- shiny::renderUI({
+      shiny::req(input$indicator)
+      shiny::tags$div(
+        title = paste("Minste antall observasjoner (N) som kreves for at",
+                      "indikatoren presenteres"),
+        shiny::numericInput(
+          ns("min_denominator"), "Minste antall observasjoner:",
+          value = rv$ind_data$min_denominator, min = 0
+        )
+      )
+    })
+
+    output$set_type <- shiny::renderUI({
+      shiny::req(input$indicator)
+      shiny::tags$div(
+        title = paste(
+          "Normalt sett vil grad av m\u00e5loppn\u00e5else gis som antall",
+          "hendelser som har oppn\u00e5dd m\u00e5let (var) delt p\u00e5",
+          "totalt antall hendelser (denominator) og da skal indikatortype",
+          "settes til 'andel' eller 'dg_andel'. Alternativt kan grad av",
+          "m\u00e5loppn\u00e5else beregnes f\u00f8r opplasting til",
+          "Sykehusviseren og da skal indikatortype settes til en av de andre",
+          "kategoriene."
+        ),
+        shiny::selectInput(
+          ns("type"), "Indikatortype:",
+          choices = unique(rv$type), selected = rv$ind_data$type
```

Comment:
Vi burde kanskje legge inn `choices` som en liste i config. Nå kan man kun velge type som allerede ligger inne for gitt register. Det vil si at hvis register kun kan `andel` på alle indikatorene sine så kan man ikke velge andre typer her.

Reply:
Helt enig, og vi bør bestemme hvilke kategorier som faktisk skal benyttes (være lov). Akkurat nå er det litt "vill-vest" i bruk av indikatortyper... Pt så kan det se ut som om vi bare trenger to typer: de som er ekte andeler og de som ikke er det (dg-andel er eksempelvis overflødig). Er det nok om vi legger opp til eksempelvis **andel** og **ferdigberegnet**. I så fall kan jeg enkelt sørge for den endringen her. Så vil det også være enkelt å legge til nye om vi ser behovet på et seinere tidspunkt

Reply:
Jeg tror `dg_andel` brukes i `qmongjs`, men vi burde gå gjennom dette og finne ut hva som er nødvendig og hva som ikke er nødvendig. `qmongjs` må vite om den skal ha en dekningsgradfane eller ikke, og hvilke indikatorer som eventuelt skal legges der. Jeg tror `dg_*` brukes til å identifisere disse.

Reply:
Da har jeg som et forslag satt fire mulige kategorier: andel, dg_andel, beregnet og dg_beregnet. Alle er definert i config. Går det an? I så fall så krever denne endringen en konvertering/oppretting av dagens kategorier for ind.type i databasen.

Reply:
Har lagt inn noen flere types her: #210